vNext/src/Microsoft.AspNet.OData/Query/ODataQueryOptions.cs

```diff
@@ -74,6 +76,22 @@ public ODataQueryOptions(ODataQueryContext context, HttpRequest request)
         public SelectExpandQueryOption SelectExpand { get; private set; }
 
         /// <summary>
+        /// Specifies a non-negative integer n that excludes the first n items of the queried collection from the result.         
+        /// </summary>
+        /// <remarks>
+        /// Corresponds to the $skip query option.  The service returns items starting at position n+1.
+        /// </remarks>
+        public int? Skip { get; private set; }
```

Comment:
The code which parses the value of $skip and $top throws an ODataException for negative values.  I prefer to avoid uints because they're not CLS compliant.  uint doubles the range of the parameter - but do we think a $skip or $top query with a value above Int32.MaxValue is particularly useful?
Date: Thu, 15 Oct 2015 11:20:38 -0700
From: notifications@github.com
To: WebApi@noreply.github.com
CC: evanmachusak@hotmail.com
Subject: Re: [WebApi] Support for beta 8 (#527)

In vNext/src/Microsoft.AspNet.OData/Query/ODataQueryOptions.cs:

> @@ -74,6 +76,22 @@ public ODataQueryOptions(ODataQueryContext context, HttpRequest request)
>          public SelectExpandQueryOption SelectExpand { get; private set; }
>  
>          /// <summary>
> +        /// Specifies a non-negative integer n that excludes the first n items of the queried collection from the result.         
> +        /// </summary>
> +        /// <remarks>
> +        /// Corresponds to the $skip query option.  The service returns items starting at position n+1.
> +        /// </remarks>
> +        public int? Skip { get; private set; }

Can we use unsigned here? does it make sense to take -3 rows?


—
Reply to this email directly or view it on GitHub.